packages/token-bridge-sdk/src/hooks/__tests__/useBalance.test.tsx

```diff
@@ -158,4 +173,116 @@ describe('useBalance', () => {
+        },
+        {
+          balance: BigNumber.from(5)
+        },
+        {
+          balance: BigNumber.from(20)
+        }
+      ])
+    )
+
+    const erc20 = [
+      '0x0000000000000000000000000000000000000000',
+      '0x0000000000000000000000000000000000000001',
+      '0x0000000000000000000000000000000000000002'
+    ]
+    const { result, waitForValueToChange } = renderHook(
+      () =>
+        useBalance({
+          provider,
+          walletAddress: '0x58b6A8A3302369DAEc383334672404Ee733aB239',
+          erc20Addresses: erc20
+        }),
+      { wrapper: Container }
+    )
+
+    await waitForValueToChange(() => result.current.erc20, { timeout: 100 })
+    const {
+      current: {
+        erc20: [erc20Balances]
+      }
+    } = result
+
+    expect(erc20Balances).toEqual({
+      '0x0000000000000000000000000000000000000000': BigNumber.from(10),
+      '0x0000000000000000000000000000000000000001': BigNumber.from(5),
+      '0x0000000000000000000000000000000000000002': BigNumber.from(20)
+    })
+  })
+
+  it('setter update ERC20 balance and merge data', async () => {
+    const provider = new StaticJsonRpcProvider(
+      process.env.REACT_APP_ETHEREUM_RPC_URL,
+      1
+    )
+    jest.spyOn(MultiCaller.prototype, 'getTokenData').mockImplementation(() =>
+      Promise.resolve([
+        {
+          balance: BigNumber.from(11)
+        },
+        {
+          balance: BigNumber.from(22)
+        }
+      ])
+    )
+
+    const erc20 = [
+      '0x0000000000000000000000000000000000000000',
+      '0x0000000000000000000000000000000000000001'
+    ]
+    const { result, waitForValueToChange } = renderHook(
+      () =>
+        useBalance({
+          provider,
+          walletAddress: '0x58b6A8A3302369DAEc383334672404Ee733aB239',
+          erc20Addresses: erc20
+        }),
+      { wrapper: Container }
+    )
+
+    await waitForValueToChange(() => result.current.erc20, { timeout: 100 })
+
+    const {
+      current: {
+        erc20: [erc20Balances, updateErc20Balances]
+      }
+    } = result
+
+    expect(erc20Balances).toEqual({
+      '0x0000000000000000000000000000000000000000': BigNumber.from(11),
+      '0x0000000000000000000000000000000000000001': BigNumber.from(22)
+    })
+
+    await act(async () => {
+      updateErc20Balances({
+        '0x0000000000000000000000000000000000000001': BigNumber.from(25),
+        '0x0000000000000000000000000000000000000002': BigNumber.from(33)
+      })
+      await waitForValueToChange(() => result.current.erc20, { timeout: 500 })
+
+      /**
+       * 0x..0 is untouched
+       * 0x..1 is updated
+       * 0x..2 is added
+       */
+      expect(result.current.erc20[0]).toEqual({
+        '0x0000000000000000000000000000000000000000': BigNumber.from(11),
+        '0x0000000000000000000000000000000000000001': BigNumber.from(25),
+        '0x0000000000000000000000000000000000000002': BigNumber.from(33)
+      })
+    })
+  })
 })
```

Comment:
maybe add tests for inputs with letters and test with mixed case and lower case?